frontend/public/components/network-policies/network-policy-form.tsx

```diff
@@ -0,0 +1,368 @@
+          value={networkPolicy.namespace}
+          id="namespace"
+          name="namespace"
+          aria-describedby="namespace-help"
+          required
+        />
+        <div className="help-block" id="namespace-help">
+          <p>{t('public~The namespace containing pods on which this network policy applies.')}</p>
+        </div>
+      </div>
+      <div className="form-group co-create-networkpolicy__podselector">
+        <Tooltip
+          content={t(
+            'public~A label selector, matching by key/value pairs. It is satisfied when the pod labels include all the provided pairs.',
+          )}
+        >
+          <span>
+            <label>{t('public~Pod selector')}</label> <InfoAltIcon />
+          </span>
+        </Tooltip>
+        <NameValueEditorComponent
+          nameValuePairs={networkPolicy.podSelector}
+          valueString={t('public~Selector')}
+          nameString={t('public~Label')}
+          addString={t('public~Add label')}
+          readOnly={false}
+          allowSorting={false}
+          updateParentData={handleMainPodSelectorChange}
+        />
+      </div>
+      <div className="form-group co-create-networkpolicy__type">
+        <Title headingLevel="h2">{t('public~Policy type')}</Title>
+        <YellowExclamationTriangleIcon className="co-icon-space-r" />
+        {t(
+          'public~When using the OpenShift SDN cluster network provider, egress network policy is not supported.',
+        )}
+      </div>
+      <div className="form-group co-create-networkpolicy__deny">
+        <label>{t('public~Select default ingress and egress deny rules')}</label>
+
+        <div style={{ display: 'flex' }}>
+          <Checkbox
+            label={t('public~Deny all ingress traffic')}
+            onChange={handleDenyAllIngress}
+            checked={networkPolicy.ingress.denyAll}
+            name="denyAllIngress"
+          />
+          <Checkbox
+            label={t('public~Deny all egress traffic')}
+            onChange={handleDenyAllEgress}
+            checked={networkPolicy.egress.denyAll}
+            name="denyAllEgress"
+          />
+        </div>
+      </div>
+      {!networkPolicy.ingress.denyAll && (
+        <ExpandableSection
+          toggleText={t('public~Ingress')}
+          onToggle={handleToggleIngress}
+          isExpanded={ingressExpanded}
+          className="co-create-networkpolicy__rules-expandable"
+        >
+          <div className="co-create-networkpolicy__header">
+            <div className="co-create-networkpolicy__header-right">
+              <Button
+                variant="link"
+                isDisabled={networkPolicy.ingress.rules.length === 0}
+                onClick={removeAllIngress}
+              >
+                {t('public~Remove all')}
+              </Button>
+            </div>
+            <Button variant="secondary" onClick={addIngressRule}>
+              {t('public~Add ingress rule')}
+            </Button>
```

Comment:
Found a usability issue during the testing,  by clicking `Add ingress rule` or `Add egress rule` button, user can add as many ingress/egress rules as wanted or accidentally which will generate some YAML files like
```
  ingress:
    - ports:
        - protocol: TCP
          port: 443
      from:
        - podSelector:
            matchLabels:
              role: frontend
    - ports:
        - protocol: SCTP
          port: 80
      from:
        - podSelector:
            matchLabels:
              role: frontend
          namespaceSelector:
            matchLabels:
              project: myproject
```

However for `spec.ingress.from` we should better generate them in a way 
```
spec:
  podSelector:
    matchLabels:
      role: db
  policyTypes:
  - Ingress
  - Egress
  ingress:
  - from:
    - ipBlock:   //only one occurance
        cidr: 172.17.0.0/16
        except:
        - 172.17.1.0/24
    - namespaceSelector:  //only one occurance
        matchLabels:
          project: myproject
    - podSelector:  //only one occurance
        matchLabels:
          role: frontend
    ports:  //only one occurance
    - protocol: TCP
      port: 6379
```
<img width="602" alt="Screen Shot 2021-05-20 at 4 51 51 PM" src="https://user-images.githubusercontent.com/12692381/118949536-ec3dd500-b98b-11eb-9873-6ee84f3668b6.png">

Reply:
User can add or update rules only in one ingress rule editor, shall we?

Reply:
As far as I know the spec allows to have any number of rules, there's no restriction about the number of occurrences of ipblocks / pods/namespaces selectors. It allows to build advanced configurations, e.g. several ip blocks or several namespaces with different labels. Though in terms of "good practices", to be honest, I don't know if it's ok, or if it's more recommended to create multiple policies. This was discussed with the networking team and we decided to allow any number of rules and peers within a policy. (cc @astoycos )

Reply:
@jotak The expand/collapse headings on Ingress, Egress and the peers shouldn't be blue. The carrot is the only clickable action and not the title. Thanks!

Reply:
@andrew-ronaldson this is defined in patternfly's expandable: https://www.patternfly.org/v4/components/expandable-section
You think I should use something different or create a new component?

Reply:
There is a component called field groups that should work.
https://www.patternfly.org/v4/components/form#field-groups

Reply:
I've replaced ExpandableSection with FormFieldGroupExpandable, here's how it looks:

- Peers collapsed:
![Capture d’écran de 2021-05-20 18-17-59](https://user-images.githubusercontent.com/2153442/119015624-5d37b980-b999-11eb-9388-89b8352ef4a1.png)

- Peers expanded:
![Capture d’écran de 2021-05-20 18-18-06](https://user-images.githubusercontent.com/2153442/119015638-6163d700-b999-11eb-8323-73919959812a.png)

At the moment I left the default styles for these form groups, which is quite different from what there was in the mockup (especially for Ingress and Egress). Considering what @yapei commented earlier (having several different font sizes), should we change them or not?

Reply:
On the ingress rule expandable section the description text below is further down than the PF component. This is a snippet from the PF site. 
`<div class="pf-c-form__field-group-header-description">Field group 1 description text.</div>`

As for the text sizing I'd like to adjust the default components so we have some hierarchy from: 

- Ingress
- - ingress rule
- - - peers